pywbem_mock/_wbemconnection_mock.py

```diff
@@ -109,94 +106,80 @@ def _uprint(dest, text):
+          :class:`~pywbem.CIMClassName` object which has its `namespace`
+          property set to the target namespace of the invocation. Its `host`
+          property will not be set.
+
+      methodname (:term:`string`):
+        The CIM method name that is being invoked. This is the method name for
+        which the callback function was registered. This parameter allows a
+        single callback function to be registered for multiple methods.
+
+      params (:ref:`NocaseDict`):
+        The input parameters for the method that were passed to the
+        `InvokeMethod` operation.
+
+        Each dictionary item represents a single input parameter for the CIM
+        method and is a :class:`~pywbem.CIMParameter` object, regardless of
+        how the input parameter was passed to the
+        :meth:`~pywbem.WBEMConnection.InvokeMethod` method.
+
+        The :class:`~pywbem.CIMParameter` object will have at least the
+        following properties set:
+
+        * name (:term:`string`): Parameter name (case independent).
+        * type (:term:`string`): CIM data type of the parameter.
+        * value (:term:`CIM data type`): Parameter value.
 
     Returns:
 
-      * The user written callback method must return a tuple
-        consisting of
-          * ReturnValue (:term:`CIM data type`):
-              Return value from the executed method.
+      tuple: The callback function must return a tuple consisting of:
+
+      * return_value (:term:`CIM data type`): Return value for the CIM
+        method.
 
-          * outparams (:term:`py:iterable` of CIMParameters):
+      * out_params (:term:`py:iterable` of :class:`~pywbem.CIMParameter`):
 
-            Each iterated item represents a single output parameter
-            for the CIM method and must be a
-            :class:`~pywbem.CIMParameter` with the
-            following data:
+        Each item represents a single output parameter of the CIM method.
+        The :class:`~pywbem.CIMParameter` objects must have at least
+        the following properties set:
 
-              * key (:term:`unicode string`):
-                Parameter name
-              * type (:term:`string`:
-                String representation of parameter type.
-              * value (:term:`CIM data type`):
-                Parameter value
+        * name (:term:`string`): Parameter name (case independent).
+        * type (:term:`string`): CIM data type of the parameter.
+        * value (:term:`CIM data type`): Parameter value.
 
     Raises:
 
-      Since the user defined callback method mocks a
-      WBEMServer method, it should return :class:`~pywbem.CIM_Error`
-      exceptions. Any other exceptions generated by the callback will be mapped
-      to :class:`~pywbem.CIM_Error` with status CIM_ERR_FAILED and
-      information on the exception including a traceback.
-    """
+      : Since the callback function mocks a CIM method invocation, it should
```

Comment:
Add back WBEM Server. "mocks a WBEM server method invocation.  The CIM did not really fit

Reply:
**Discussion point:** It is a CIM method, i.e. a method defined in a CIM class. I wanted to call this out in contrast to Python methods.

Reply:
You are right. Forget comment